rpc/dial.go

```diff
@@ -322,6 +322,9 @@ func listMulticastInterfaces() []net.Interface {
 	return interfaces
 }
 
+// ErrMDNSQueryFailed is returned when a mDNS query fails to find a candidate.
+var ErrMDNSQuery = errors.New("mDNS query failed to find a candidate")
```

Comment:
this is not a very descriptive name - can we name this `ErrMDNSNoCandidatesFound` or something in that vein?